registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java

```diff
@@ -577,13 +577,7 @@ private String getUUID() {
 	private String getDataShareUrl(String id, String process) throws Exception {
 		Map<String, List<String>> typeAndSubtypMap = createTypeSubtypeMapping();
 		List<String> modalities = new ArrayList<>();
-		for (Map.Entry<String, List<String>> entry : typeAndSubtypMap.entrySet()) {
-			if (entry.getValue() == null) {
-				modalities.add(entry.getKey());
-			} else {
-				modalities.addAll(entry.getValue());
-			}
-		}
+		modalities.addAll(typeAndSubtypMap.keySet());

```

Comment:
Since the same object is passed to priorityBasedPacketManagerService, sub type filter from policy will not be taken into account while sharing the biometrics to abis. This is not correct. We can create create a separate combined object for that and use.

Reply:
fixed